app/graphql/mutations/create_event_mutation.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'mutation_root'
+require_relative 'transformers/panoptes_classification'
+
+module Mutations
+  class CreateEventMutation < Mutations::BaseMutation
+    null true
+
+    argument :event_payload, [String], required: true, description: 'list of events in json format'
+
+    field :errors, [String], null: false
+
+    def resolve(event_payload:)
+      # TODO: Add guard here to only ingest data that meets the schema
+      # avoid ouroboros data and panoptes talk data until we have verified
+      # the schema conformance here
+      # return unless model.type && model.source == panoptes classificaiton
+      
+      # TODO: authorisation here
+      if event_payload.nil?
+        return {errors: [{"message" => "ArgumentError"}]}
+      end
+      
+      events_list = []
+      event_payload.each do |event|
+        event_hash = eval(event)
```

Comment:
Be careful with eval on user submitted data - it can open up code execution vectors for specially crafter payloads. https://stackoverflow.com/questions/637421/is-eval-supposed-to-be-nasty

Instead look at using JSON parsing, e.g. https://github.com/zooniverse/interventions-gateway/blob/1afd121662ac255577d86ccc58b4dace5dd0bdf1/interventions_gateway_api.rb#L43

Currently you have an array of JSON strings, i'd assume the payload will be a JSON string representing an array on objects? 
```
let(:event_payload) { [{test: "hash"}].to_json } 
# "[{\"test\":\"hash\"}]" vs ["{\"test\" => \"hash\"}"] 
```
The system posting data in will setup a batch of event objects in an array and convert that array to json string before POSTing data. As such you can call `JSON.parse(payload)` to convert the string back to an array of objects (hashes).

Reply:
Never trust the submitted input - https://www.owasp.org/index.php/Don%27t_trust_user_input

Please switch to using a `JSON.parse` as all incoming payloads should be `content-type=application/json`. This will allow us to capture errors with incoming json formatting too, either we allow the 500 and report it in rollbar or you capture that explicit case and add a 400 HTTP response error with a meaningful message.

Reply:
And another, possibly even more technical, reason why you shouldn't do this: `{"foo": "bar"}` means that the key `foo` will be a symbol rather than a string. In Ruby, symbols defined directly in code (and thus, via eval) won't be garbage collected, meaning they'll continue to consume memory forever, even when the hash they were used in is no longer in memory itself.

This means an attacker could also cause your application to run out of memory by sending in lots of events with random keys in them. For that, they wouldn't even need to hack Panoptes, since they can just send in classifications with random keys in the annotations.

Reply:
i think the symbol leak got fixed in ruby GC in 2.2 https://www.sitepoint.com/symbol-gc-ruby-2-2/

Reply:
Ones defined in code literally still don't get GCd (as in the article) and I checked that evalled ones don't either (using the same code from that article).

Reply:
good point - they are eval'd